src/backend/optimizer/util/pathnode.c

```diff
@@ -4056,6 +4056,36 @@ create_hashjoin_path(PlannerInfo *root,
 	if (CdbPathLocus_IsNull(join_locus))
 		return NULL;
 
+	/*
+	 * We check if our path will be rescanned due to parametrization
+	 * from outer path after this node.
+	 * For all information see the comments in create_nestloop_path().
```

Comment:
Maybe it would be better to copy all the comments from there? Or even move the shared section of code into a separate function?